clang/test/SemaSYCL/num_simd_work_items_device.cpp

```diff
@@ -42,66 +42,88 @@ struct FuncObj {
-  [[intel::reqd_work_group_size(5)]] //expected-note{{conflicting attribute is here}}
+  [[intel::num_simd_work_items(4)]]      // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+  [[intel::reqd_work_group_size(4, 64)]] // expected-note{{conflicting attribute is here}}
   void
   operator()() const {}
 };
 
 struct TRIFuncObjBad6 {
-  [[intel::reqd_work_group_size(5)]] // expected-note{{conflicting attribute is here}}
-  [[intel::num_simd_work_items(3)]]  // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+  [[intel::reqd_work_group_size(4, 64)]] // expected-note{{conflicting attribute is here}}
+  [[intel::num_simd_work_items(4)]]      // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
   void
   operator()() const {}
 };
 
 struct TRIFuncObjBad7 {
-  [[intel::num_simd_work_items(4)]]      // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
-  [[intel::reqd_work_group_size(3, 64)]] // expected-note{{conflicting attribute is here}}
+  [[cl::reqd_work_group_size(6, 3, 5)]] // expected-note{{conflicting attribute is here}}
+  [[intel::num_simd_work_items(3)]]     // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
   void
   operator()() const {}
 };
 
 struct TRIFuncObjBad8 {
-  [[intel::reqd_work_group_size(3, 64)]] // expected-note{{conflicting attribute is here}}
-  [[intel::num_simd_work_items(4)]]      // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+  [[intel::num_simd_work_items(3)]]     // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+  [[cl::reqd_work_group_size(6, 3, 5)]] // expected-note{{conflicting attribute is here}}
   void
   operator()() const {}
 };
 
+[[intel::num_simd_work_items(2)]] // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+[[intel::reqd_work_group_size(4, 2, 3)]] void func1(); // expected-note{{conflicting attribute is here}}
+
+[[intel::reqd_work_group_size(4, 2, 3)]] // expected-note{{conflicting attribute is here}}
+[[intel::num_simd_work_items(2)]] void func2(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+[[intel::num_simd_work_items(2)]] // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+[[cl::reqd_work_group_size(4, 2, 3)]] void func3(); // expected-note{{conflicting attribute is here}}
```

Comment:
This one *might* be wrong once we figure out the final answer to the `cl` spelling.

Reply:
> This one _might_ be wrong once we figure out the final answer to the `cl` spelling.

FYI : https://github.com/intel/llvm/pull/3728#discussion_r631161682

Reply:
> This one _might_ be wrong once we figure out the final answer to the `cl` spelling.

No change in the latest patch that follows the current discussion of cl::reqd_work_group_size(Z,Y,X).
 Will update based on the final answer.

Reply:
> This one _might_ be wrong once we figure out the final answer to the `cl` spelling.

Thanks @gmlueck and @AaronBallman for the discussion here https://github.com/intel/llvm/pull/3728#discussion_r631382049

Based on the inputs (i agree that  SYCL 1.2.1 spec as well as SYCL 2020 is very unclear about the meaning of this attribute parameters), i did not make any change to the cl spelling. The current PR is  interpreting attribute parameters as (z, y, x), where x varies fastest.